packages/x-date-pickers/src/PickersDay/PickersDay.tsx

```diff
@@ -112,7 +112,7 @@ const useUtilityClasses = (ownerState: PickersDayProps<any>) => {
   const slots = {
     root: [
       'root',
-      selected && 'selected',
+      selected && (!outsideCurrentMonth || showDaysOutsideCurrentMonth) && 'selected',
```

Comment:
It might be easier to read/hunderstand if you define `isHiddenDaySpacingFiller = outsideCurrentMonth && !showDaysOutsideCurrentMonth` before `slots`

```suggestion
      selected && !isHiddenDaySpacingFiller && 'selected',
```

Reply:
Thank you for a great suggestion! ❤️